app/controllers/tracks_controller.rb

```diff
@@ -78,6 +78,13 @@ def audio
     end
   end
 
+  def download
+    audio_file = @track.audio_file
+    raise ActiveRecord::RecordNotFound.new('track has no audio', 'audio') unless audio_file.present? && audio_file.check_self
+
+    send_file audio_file.full_path
```

Comment:
I'd add:
```suggestion
    send_file audio_file.full_path, filename: audio_file.filename, disposition: :attachment
```

Reply:
Maybe even edit the filename so that it contains the codec's extension? Systems can behave differently when files contain the correct mime headers but no extension.

Reply:
According to [the docs](https://api.rubyonrails.org/classes/ActionController/DataStreaming.html#method-i-send_file), the default `:filename` is the basename of the file on the file system. `disposition: "attachment" ` is also the default.

Reply:
Sorry, I was thinking of `send_blob`